test/e2e/e2e_test.go

```diff
@@ -79,4 +80,38 @@ var _ = ginkgo.Describe("TrainJob e2e", func() {
 			})
 		})
 	})
+
+	ginkgo.When("should succeed to perform OpenMPI workload", func() {
+		ginkgo.It("should create TrainJob with OpenMPI runtime reference", func() {
+			trainJob := testingutil.MakeTrainJobWrapper(ns.Name, "e2e-test-openmpi").
+				RuntimeRef(trainer.SchemeGroupVersion.WithKind(trainer.ClusterTrainingRuntimeKind), mpiRuntime).
+				Obj()
+
+			ginkgo.By("Create a TrainJob with mpi-distributed runtime reference", func() {
+				gomega.Expect(k8sClient.Create(ctx, trainJob)).Should(gomega.Succeed())
+			})
+
+			// Wait for TrainJob to be in Succeeded status.
+			ginkgo.By("Wait for TrainJob to be in Succeeded status", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					gotTrainJob := &trainer.TrainJob{}
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(trainJob), gotTrainJob)).Should(gomega.Succeed())
+					g.Expect(gotTrainJob.Status.Conditions).Should(gomega.BeComparableTo([]metav1.Condition{
+						{
+							Type:    trainer.TrainJobCreated,
+							Status:  metav1.ConditionTrue,
+							Reason:  trainer.TrainJobJobsCreationSucceededReason,
+							Message: constants.TrainJobJobsCreationSucceededMessage,
+						},
+						{
+							Type:    trainer.TrainJobComplete,
+							Status:  metav1.ConditionTrue,
+							Reason:  jobsetconsts.AllJobsCompletedReason,
+							Message: jobsetconsts.AllJobsCompletedMessage,
+						},
+					}, util.IgnoreConditions))
```

Comment:
Do we want to move this to util function ?
E.g. `TrainJobComplete`

Reply:
I do not want to do it. For production codes, yes, based on the DRY principle.
However, this should be DUMP since this is testing codes..